### PR TITLE
Pass configuration to showBuildSettings command

### PR DESCRIPTION
**Pull request by maschall**
#6113

### CODE REVIEW

File: fastlane_core/spec/project_spec.rb [hunk 1]
```diff
@@ -268,7 +268,7 @@ def within_a_temp_dir
       end
     end
 
-    describe "Build Settings" do
+    describe "Build Settings with defualt configuration" do
```
Comment (KrauseFx) [inline fastlane_core/spec/project_spec.rb]: Should be `default`

**Review by KrauseFx — DISMISSED** [review]
Looks good, thanks for your PR and the tests. Does this change behaviour for existing projects and setups? How can a user start using this?

File: fastlane_core/spec/project_spec.rb (hunk 1 above)
Comment (maschall) [inline fastlane_core/spec/project_spec.rb]: fixed, thanks!

**Comment by maschall** [comment]
No the change will not effect anyone, because the configuration defaults based on the Xcode project settings.



So status quo is kept.

I don't see any actions that directly report back build_settings, but ultimately the example that I need this for, is that we have a project with multiple app ids for different configurations (enterprise vs app store, because of Push Notifications).  I want to pass the app_indentifier to sigh, so I can do the following with this change.

```
app_identifier = FastlaneCore::Project.new(workspace: test.xcworkspace, scheme: myScheme configuration: Enterprise).build_settings(key: "PRODUCT_BUNDLE_IDENTIFIER")
sigh( app_identifier: app_identifier )
```

I'm not sure if it exists already, but it'd be nice if there was a way to set workspace, scheme, configuration in one spot and use it in the actions (probably with lane_context) and then a Fastfile could have something like:

```
lane :build do |options|
   lane_context[SharedValues::SCHEME] = options[:scheme]
   lane_context[SharedValues::WORKSPACE] = options[:workspace]  
   lane_context[SharedValues::CONFIGURATION] = options[:configuration]
   
   app_identifier = get_app_identifier
   sigh(app_identifier: app_identifier)
   gym
end
```

Obviously my example is because our project is pretty complex with 6 schemes, with 4 configurations each, so Gymfiles and Appfiles get difficult to use.

**Comment by maschall** [comment]
@KrauseFx *bump* Just seeing if the change is resolved.

**Review by milch — APPROVED** [review]
Looks good to me, I don't think anything should break here for users. 

@KrauseFx How do you feel about merging this?

**Comment by milch** [comment]
Re-tested after rebasing on current master, still seems to work. 

Thanks a lot for your contribution @maschall, and sorry for the long delay 🚀

**Comment by mfurtak** [comment]
This appears to be causing #7106.

#7107, #7108, and #7109 seem like sensible fixes, so I'll push those through.

**Pull request merged**